src/main.cpp

```diff
@@ -4840,14 +4852,21 @@ bool ProcessBlock(CNode* pfrom, CBlock* pblock, bool generated_by_me)
              mi != mapOrphanBlocksByPrev.upper_bound(hashPrev);
              ++mi)
         {
-            CBlock* pblockOrphan = (*mi).second;
+            CBlock* pblockOrphan = mi->second;
             if (pblockOrphan->AcceptBlock(generated_by_me))
                 vWorkQueue.push_back(pblockOrphan->GetHash());
+            delete mapOrphanBlocks.at(pblockOrphan->GetHash());
             mapOrphanBlocks.erase(pblockOrphan->GetHash());
             setStakeSeenOrphan.erase(pblockOrphan->GetProofOfStake());
             delete pblockOrphan;
         }
+
+        auto it = mapOrphanBlocksByPrev.begin();
+        while((it = mapOrphanBlocksByPrev.find(hashPrev)) != mapOrphanBlocksByPrev.end()) {
+            delete it->second;
```

Comment:
Why this is not in `ClearOrphanBlocks` function?

Reply:
Because this code removes all orphan blocks with one specific previous block hash, it doesn't clear the entire map. It is needed only here, but could eventually be moved to a function if needed somewhere else. `ClearOrphanBlocks` clears the entire map regardless of their previous block hash